mongodb_exporter.go

```diff
@@ -38,47 +38,49 @@ const (
+	tlsPrivateKeyF = app.Flag("mongodb.tls-private-key", "Path to PEM file that contains the decrypted private key (if not contained in mongodb.tls-cert file).").Default("").String()
+	tlsCAF         = app.Flag("mongodb.tls-ca", "Path to PEM file that contains the CAs that are trusted for server connections.\n"+
 		"    \tIf provided: MongoDB servers connecting to should present a certificate signed by one of this CAs.\n"+
 		"    \tIf not provided: System default CAs are used.").Default("").String()
-	tlsDisableHostnameValidationF = kingpin.Flag("mongodb.tls-disable-hostname-validation", "Disable hostname validation for server connection.").Bool()
-	maxConnectionsF               = kingpin.Flag("mongodb.max-connections", "Max number of pooled connections to the database.").Default("1").Int()
-	testF                         = kingpin.Flag("test", "Check MongoDB connection, print buildInfo() information and exit.").Bool()
+	tlsDisableHostnameValidationF = app.Flag("mongodb.tls-disable-hostname-validation", "Disable hostname validation for server connection.").Bool()
+	maxConnectionsF               = app.Flag("mongodb.max-connections", "Max number of pooled connections to the database.").Default("1").Int()
+	testF                         = app.Flag("test", "Check MongoDB connection, print buildInfo() information and exit.").Bool()
 
-	socketTimeoutF = kingpin.Flag("mongodb.socket-timeout", "Amount of time to wait for a non-responding socket to the database before it is forcefully closed.\n"+
+	socketTimeoutF = app.Flag("mongodb.socket-timeout", "Amount of time to wait for a non-responding socket to the database before it is forcefully closed.\n"+
 		"    \tValid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'.").Default("3s").Duration()
-	syncTimeoutF = kingpin.Flag("mongodb.sync-timeout", "Amount of time an operation with this session will wait before returning an error in case\n"+
+	syncTimeoutF = app.Flag("mongodb.sync-timeout", "Amount of time an operation with this session will wait before returning an error in case\n"+
 		"    \ta connection to a usable server can't be established.\n"+
 		"    \tValid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'.").Default("1m").Duration()
 
 	// FIXME currently ignored
 	// enabledGroupsFlag = flag.String("groups.enabled", "asserts,durability,background_flushing,connections,extra_info,global_lock,index_counters,network,op_counters,op_counters_repl,memory,locks,metrics", "Comma-separated list of groups to use, for more info see: docs.mongodb.org/manual/reference/command/serverStatus/")
-	enabledGroupsFlag = kingpin.Flag("groups.enabled", "Currently ignored").Default("").String()
+	enabledGroupsFlag = app.Flag("groups.enabled", "Currently ignored").Default("").String()
 )
 
 func main() {
-	initVersionInfo()
-	kingpin.Parse()
+	initVersionInfo(app)
+	log.AddFlags(app)
```

Comment:
Can we use `log.AddFlags(kingpin.CommandLine)` and avoid most of the changes in this PR?

Reply:
I prefer to remove a global state when and where it can be removed because it, creates one of the most dangerous indirection points. But ok, as you want.